### ISSUE TITLE
Gradle Plugin Marker is published without GMM: metadataSources.gradleMetadata fails

### ISSUE THREAD

**Issue by hfhbd** [gradle/gradle#31136]

**Comment by hfhbd** [comment]
Yeah, I learned it in the hard way that you need a component (and a SoftwareComponentFactory) to create the GMM file. Registering a maven publication with `artifact(myTask)` does not create a GMM file.

Are you open for a PR?

**Comment by reinsch82** [comment]
@gradle/bt-jvm Hi Folks, can you reply to the poster whether we would be open for a PR on this issue?

**Comment by jvandort** [comment]
Can you post a reproducer on the _publication side_ for a publication of a plugin that does not produce a POM?

> Registering a maven publication with artifact(myTask) does not create a GMM file.

This seems about right

> Are you open for a PR?

What do you plan to change? Before you put in the effort perhaps its worth discussing a solution

From my POV, this code seem like it needs to be converted to use a component: 
https://github.com/gradle/gradle/blob/82937b89af179043f34052cd5d6a8818571cca8f/platforms/extensibility/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/MavenPluginPublishPlugin.java#L95-L112
https://github.com/gradle/gradle/blob/82937b89af179043f34052cd5d6a8818571cca8f/platforms/extensibility/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/IvyPluginPublishingPlugin.java#L102-L130
